control-plane/pkg/reconciler/channel/channel.go

```diff
@@ -286,7 +292,41 @@ func (r *Reconciler) reconcileKind(ctx context.Context, channel *messagingv1beta
 		return err
 	}
 
-	address, _ := url.Parse(fmt.Sprintf("http://%s.%s.svc.%s", channelService.Name, channelService.Namespace, network.GetClusterDomainName()))
+	transportEncryptionFlags := feature.FromContext(ctx)
+	if transportEncryptionFlags.IsPermissiveTransportEncryption() {
+		caCerts, err := r.getCaCerts()
+		if err != nil {
+			return err
+		}
+
+		httpAddress := r.httpAddress(channelService)
+		httpsAddress := r.httpsAddress(caCerts, channelService)
+		// Permissive mode:
+		// - status.address http address with path-based routing
+		// - status.addresses:
+		//   - https address with path-based routing
+		//   - http address with path-based routing
+		channel.Status.Addresses = []pkgduckv1.Addressable{httpsAddress, httpAddress}
+		channel.Status.Address = &httpAddress
+	} else if transportEncryptionFlags.IsStrictTransportEncryption() {
+		// Strict mode: (only https addresses)
+		// - status.address https address with path-based routing
+		// - status.addresses:
+		//   - https address with path-based routing
+		caCerts, err := r.getCaCerts()
+		if err != nil {
+			return err
+		}
+
+		httpsAddress := r.httpsAddress(caCerts, channelService)
+		channel.Status.Addresses = []pkgduckv1.Addressable{httpsAddress}
+		channel.Status.Address = &httpsAddress
+	} else {
+		httpAddress := r.httpAddress(channelService)
+		channel.Status.Address = &httpAddress
+	}
```

Comment:
We need to set the address only after the prober is successful, so that if it fails, we don't set the address which could mislead the clients to assume the addressable is ready